### PR TITLE
fix: render error UI if no Git information is found

### PR DESCRIPTION
**Pull request by mattrothenberg**
Fixes an issue where, if you run the initialize command in a directory **that is not a valid Git repository with an upstream remote**, the webview loading process hangs forever.

Instead, if this scenario occurs, I've opted to render a UI that informs the user they need a valid Git repository + remote upstream URL.

Once they've added such details, they can use the **Refresh** button to reload the webview, or manually do it as well.

Screen Shot 2022-02-01 at 4 49 16 PM

### CODE REVIEW

File: src/flatConfigEditor.ts [hunk 1]
```diff
@@ -406,7 +426,7 @@ export class FlatConfigEditor implements vscode.CustomTextEditorProvider {
 
         resolve({ name, owner })
       } catch (e) {
-        resolve({})
```
Comment (mattrothenberg) [inline src/flatConfigEditor.ts:409]: lol whyyyyyyy did I do this

File: src/git.ts [hunk 2]
```diff
@@ -31,26 +31,24 @@ export class VSCodeGit {
...
+            if (count === times) {
+              clearInterval(interval)
               reject(new Error("Couldn't get repo details"))
             }
             count++
           }
         } catch (e) {
           reject(new Error("Couldn't get repo details"))
-        } finally {
-          clearInterval(checkRepoExists)
```
Comment (mattrothenberg) [inline src/git.ts:53]: also WHY did I do this? This immediately cleared the interval, causing this promise to hang forever

File: src/flatConfigEditor.ts [hunk 3]
```diff
@@ -74,9 +74,20 @@ export class FlatConfigEditor implements vscode.CustomTextEditorProvider {
...
+        webviewPanel.webview
+      )
+    } catch (e) {
+      await vscode.window.showErrorMessage(
+        "Please make sure you're in a repository with a valid upstream GitHub remote"
+      )
+      await vscode.commands.executeCommand(
+        'workbench.action.revertAndCloseActiveEditor'
+      )
+      // For whateever reason, this doesn't close the webview.
```
Comment (Wattenberger) [inline src/flatConfigEditor.ts:88]: tbh I prefer the `whateever` spelling

File: src/git.ts [hunk 4]
```diff
@@ -31,26 +31,24 @@ export class VSCodeGit {
...
           } else {
-            if (timeToStop) {
+            if (count === times) {
+              clearInterval(interval)
               reject(new Error("Couldn't get repo details"))
             }
             count++
           }
         } catch (e) {
           reject(new Error("Couldn't get repo details"))
```
Comment (Wattenberger) [inline src/git.ts:51]: do we need to clear the interval here?

**Review by Wattenberger — APPROVED** [review]
thanks for catching that!!

File: src/git.ts (hunk 4 above)
Comment (mattrothenberg) [reply src/git.ts:51]: I'm not entirely sure tbh

**Pull request merged**